### DISCUSSION TITLE
How do I trace the syscalls made by a particular process in a container?

### DISCUSSION THREAD

**Discussion by sbmthakur** [bpftrace/bpftrace#2213] [Q&A] [answered]
`bpftrace` will be running on the host. As described [here](https://github.com/iovisor/bpftrace/discussions/1988), we can trace the system calls issued from a container. However, I want to trace the calls only for a specific process. Is there any way to do it with bpftrace?

**Comment by fbs** [comment, marked as answer]
You can attach to a specific proceses with the `-p` flag or use a `/pid == 1234/` like filter in the program. Just make sure you use the pid as seen from the NS bpftrace runs in and not the pid in the pid namespace the process runs in

**Reply by sbmthakur** [reply]
Thanks for the answer. I have concluded that [tracee](https://github.com/aquasecurity/tracee) is better suited to my needs.
